### DISCUSSION TITLE
Exporting task description edit history

### DISCUSSION THREAD

**Discussion by mskiptr** [plswiderski/asanaexporter#21] [Q&A] [answered]
Hello! Recently I was trying to export all the changes made to some Asana tasks. The official export option does not do that – even though all these modifications are recorded and displayed in the web interface. Because of that I first tried to achieve that with web scraping and after a bit of reconnaissance just wrote a tool that sifts through the WebSocket replies from the backend. You can take a look at it [here](https://git.sr.ht/~mskiptr/asana-history-extractor).

Sadly, this method loses most of the formatting and details, introduces a bit of noise (due to a bug in Asana) and only works for just one task at a time. Would you happen to know if this could be done better using the official API?

**Comment by plswiderski** [comment, marked as answer]
Hi! I don't think it is possible to obtain the whole history of the task. Just like I checked out briefly only basic information can be received from GET task request. You can get more info about dependencies and other values for tasks added via premium features.
Here is documentation: https://developers.asana.com/reference/gettask

I think the most appropriate place to ask such question would be asana's forum. https://forum.asana.com/

Have a nice day.
